cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/Index.java

```diff
@@ -61,7 +61,7 @@ private Index(List<Object> fieldNames,
      * This method sets the index type to the default setting of "json"
      *
      * @param fieldNames the field names in the index
-     * @param indexName the index name
+     * @param indexName the index name or null
```

Comment:
I'd like to see this description (and on line 80) extended to explain what happens in the `null` case, something like `or null to generate a unique index name`